OneCommunityGlobal/HGNRest
# Xiaow task system improvement

# Description
This PR aims to cooperate with frontend to overhaul the task system logic and improve UI. Major changes include but not limited to: overhaul Task component and task controller display logic, improve logic of task import, addition, move, delete and edit,  improve task reducer logic, and make code more organized.

## Related PRS (if any):
To test this backend PR you need to checkout the [#1032](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1032#issue-1816682914) frontend PR.


## Main changes explained:
- Optimize the logic of importTask, postTask, deleteTask controllers 
- Make postTask controller to modify current wbs's modifiedDatetime
- Remove updatedChildrenQty and its corresponding router,  merge its logic in deleteTask controller since those two requests are always sent at the same time in the frontend.
- Rewrite moveTask logic and make child tasks move with their mother
- Add one getWBSId controller and its corresponding router for sending wbs info to frontend for checking if new task is added to database since the page load.
…

## How to test:
1. check into current branch and frontend PR [#1032](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1032#issue-1816682914)
2. do `npm install` `npm run build` and `npm start` to run this PR locally
3. check frontend PR for further instruction
